src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1210,51 +1422,46 @@ def random_mac(self):
 
         return vnc_web
 
+    def __del__(self):
+        if self.libvirt_connection:
+            with contextlib.suppress(libvirt.libvirtError):
+                self.libvirt_connection.close()
+
+    @private
+    async def wait_for_libvirtd(self, timeout):
+        async def libvirtd_started(middleware):
+            while not await middleware.call('service.started', 'libvirtd'):
+                pass
+
+        try:
+            await asyncio.wait_for(libvirtd_started(self.middleware), timeout=timeout)
+        except asyncio.TimeoutError:
+            # TODO: We can log here but there is no good way to distinguish between a boot and middlewared restart
```

Comment:
`system.ready`?

Reply:
I had a peek at `system.ready` yesterday and it wasn't working as desired for me i.e on middlewared restarts it still showed the state as `BOOTING`, but i can't reproduce that anymore, so added a todo initially. I'll have this updated.

Reply:
> i.e on middlewared restarts it still showed the state as `BOOTING`

This should not happen, if it does, it is a bug

Reply:
I'll keep an eye out for it, i wasn't sure if that is known. Thank you for the update